mobify/progressive-web-scaffold
# Checkout Cart, Shipping & Payment

Add initial "mock" templates for Cart, Shipping and Payment pages

 **JIRA**: [WEB-1014](https://mobify.atlassian.net/browse/WEB-1014), [WEB-1015](https://mobify.atlassian.net/browse/WEB-1015)
**Mocks**: [Cart mock](https://mobify.invisionapp.com/d/main/#/console/9834411/210214375/inspect), [Shipping mock](https://mobify.invisionapp.com/d/main/#/console/9834411/210229994/inspect), [Payment mock](https://mobify.invisionapp.com/d/main/#/console/9834411/210236445/inspect)

## Changes
- Add mock template for:  Cart, Shipping and Cart page

## Todos
- [x] @kschmidtdev Notification management (#209)
- [x] ~~Add Show/Hide component~~ Make the Company/Apt # fields show/hide-able via actions
- [x] @jeffkamo Update icons (waiting on #207)
- [x] @kschmidtdev Replace hardcoded partials with (mock) data
    - More work to be done here, but can be done when we combine the front and back ends
- ~~Add custom country select~~ (maybe as additional capacity)

### Cart Page
- [x] @jeffkamo Fix field layout issue (`align-items: baseline`)
- [x] Add modals (Shopping + Wishlist)
- [x] Add empty cart state
- [x] @jeffkamo Add placeholder skeletons

### Shipping Page
- [x] Remove "Email Receipt to" heading text
- [x] Add "Welcome back" notification styles (depends on #209)
- [x] Add states for login password
- [x] Add states for new shipping address

### ~~Payment Page~~
Payment page is out of scope for this sprint (additional capacity)
- [x] Add states for new credit cart
- [x] Add states for new billing address
- ~~Add fixed "Place Order" container~~
- ~~Add credit card component~~

### Last Stuff
- [x] Confirm with @kschmidtdev that tests are adequate
    - There are now enough tests to pass the build
    - Not enough tests to be truly adequate to cover all edge cases or possible issues
- [x] Confirm nothing links to this new cart/checkout flow
    - There is a link in the mini-cart and PDP "added to cart" modal
- [x] Test on desktop
- [x] Test on tablet
- [ ] @mikenikles to review

## How to test-drive this PR
- Pull down this branch
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Add any product to the cart
- Visit the [cart page](https://www.merlinspotions.com/checkout/cart/)
- Visit the [shipping page](https://www.merlinspotions.com/checkout/shipping/)
- Ensure that the design matches the mockups
